src/fsharp/service/service.fs

```diff
@@ -518,14 +518,26 @@ type TypeCheckInfo
         | Item.Types(_, ty::_) when (isInterfaceTy g ty) -> true
         | _ -> false   
 
+
     // Return only items with the specified name
     let FilterDeclItemsByResidue (getItem: 'a -> Item) residue (items: 'a list) = 
+        let attributedResidue = residue + "Attribute"
+        let nameMatchesResidue name = residue = name || attributedResidue = name
+
         items |> List.filter (fun x -> 
             let item = getItem x
             let n1 =  item.DisplayName 
             match item with
-            | Item.Types _ | Item.CtorGroup _ -> residue + "Attribute" = n1 || residue = n1
-            | _ -> residue = n1 )
+            | Item.Types _ -> nameMatchesResidue n1
+            | Item.CtorGroup (_, meths) ->
+                nameMatchesResidue n1 ||
+                meths |> List.exists (fun meth ->
+                    match meth.ApparentEnclosingAppType with
```

Comment:
Use `meth.ApparentEnclosingTyconRef` to avoid the need for the `match`

Also use `tcref` not `tyref`